# Upgrades Flux image-automation-controller and image-reflector-controller CRDs
# from v0.x to v1.1.0, promoting storage versions from v1beta2 to v1 (GA).
# v1beta2 is retained but marked deprecated. Adds new fields (policySelector,
# digestReflectionPolicy, proxySecretRef, etc.) and improves printer columns.
# - https://github.com/chainguard-dev/customer-issues/issues/2972
diff -ru a/bitnami/flux/templates/crds/image-automation-controller-crds.yaml b/bitnami/flux/templates/crds/image-automation-controller-crds.yaml
--- a/bitnami/flux/templates/crds/image-automation-controller-crds.yaml	2026-02-20 01:24:48
+++ b/bitnami/flux/templates/crds/image-automation-controller-crds.yaml	2026-02-20 01:29:43
@@ -1,13 +1,14 @@
 # Source: https://github.com/fluxcd/image-automation-controller/releases/download/v{version}/image-automation-controller.crds.yaml
-# Version: 0.41.2
+# Version: 1.1.0
 # VersionOf: fluxcd-image-automation-controller
 # Conditional: .Values.imageAutomationController.installCRDs
 {{- if .Values.imageAutomationController.installCRDs }}
+---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.16.1
+    controller-gen.kubebuilder.io/version: v0.19.0
   name: imageupdateautomations.image.toolkit.fluxcd.io
 spec:
   group: image.toolkit.fluxcd.io
@@ -15,16 +16,28 @@
     kind: ImageUpdateAutomation
     listKind: ImageUpdateAutomationList
     plural: imageupdateautomations
+    shortNames:
+    - iua
+    - imgupd
+    - imgauto
     singular: imageupdateautomation
   scope: Namespaced
   versions:
   - additionalPrinterColumns:
+    - jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].message
+      name: Status
+      type: string
     - jsonPath: .status.lastAutomationRunTime
       name: Last run
+      priority: 1
       type: string
-    deprecated: true
-    deprecationWarning: v1beta1 ImageUpdateAutomation is deprecated, upgrade to v1beta2
-    name: v1beta1
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v1
     schema:
       openAPIV3Schema:
         description: ImageUpdateAutomation is the Schema for the imageupdateautomations
@@ -119,7 +132,15 @@
                         description: |-
                           MessageTemplate provides a template for the commit message,
                           into which will be interpolated the details of the change made.
+                          Note: The `Updated` template field has been removed. Use `Changed` instead.
                         type: string
+                      messageTemplateValues:
+                        additionalProperties:
+                          type: string
+                        description: |-
+                          MessageTemplateValues provides additional values to be available to the
+                          templating rendering.
+                        type: object
                       signingKey:
                         description: SigningKey provides the option to sign commits
                           with a GPG key
@@ -181,6 +202,54 @@
                   run should be attempted.
                 pattern: ^([0-9]+(\.[0-9]+)?(ms|s|m|h))+$
                 type: string
+              policySelector:
+                description: |-
+                  PolicySelector allows to filter applied policies based on labels.
+                  By default includes all policies in namespace.
+                properties:
+                  matchExpressions:
+                    description: matchExpressions is a list of label selector requirements.
+                      The requirements are ANDed.
+                    items:
+                      description: |-
+                        A label selector requirement is a selector that contains values, a key, and an operator that
+                        relates the key and values.
+                      properties:
+                        key:
+                          description: key is the label key that the selector applies
+                            to.
+                          type: string
+                        operator:
+                          description: |-
+                            operator represents a key's relationship to a set of values.
+                            Valid operators are In, NotIn, Exists and DoesNotExist.
+                          type: string
+                        values:
+                          description: |-
+                            values is an array of string values. If the operator is In or NotIn,
+                            the values array must be non-empty. If the operator is Exists or DoesNotExist,
+                            the values array must be empty. This array is replaced during a strategic
+                            merge patch.
+                          items:
+                            type: string
+                          type: array
+                          x-kubernetes-list-type: atomic
+                      required:
+                      - key
+                      - operator
+                      type: object
+                    type: array
+                    x-kubernetes-list-type: atomic
+                  matchLabels:
+                    additionalProperties:
+                      type: string
+                    description: |-
+                      matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
+                      map is equivalent to an element of matchExpressions, whose key field is "key", the
+                      operator is "In", and the values array contains only "value". The requirements are ANDed.
+                    type: object
+                type: object
+                x-kubernetes-map-type: atomic
               sourceRef:
                 description: |-
                   SourceRef refers to the resource giving access details
@@ -231,8 +300,6 @@
                     enum:
                     - Setters
                     type: string
-                required:
-                - strategy
                 type: object
             required:
             - interval
@@ -325,16 +392,55 @@
               observedGeneration:
                 format: int64
                 type: integer
+              observedPolicies:
+                additionalProperties:
+                  description: ImageRef represents an image reference.
+                  properties:
+                    digest:
+                      description: Digest is the image's digest.
+                      type: string
+                    name:
+                      description: Name is the bare image's name.
+                      type: string
+                    tag:
+                      description: Tag is the image's tag.
+                      type: string
+                  required:
+                  - name
+                  - tag
+                  type: object
+                description: |-
+                  ObservedPolicies is the list of observed ImagePolicies that were
+                  considered by the ImageUpdateAutomation update process.
+                type: object
+              observedSourceRevision:
+                description: |-
+                  ObservedPolicies []ObservedPolicy `json:"observedPolicies,omitempty"`
+                  ObservedSourceRevision is the last observed source revision. This can be
+                  used to determine if the source has been updated since last observation.
+                type: string
             type: object
         type: object
     served: true
-    storage: false
+    storage: true
     subresources:
       status: {}
   - additionalPrinterColumns:
+    - jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].message
+      name: Status
+      type: string
     - jsonPath: .status.lastAutomationRunTime
       name: Last run
+      priority: 1
       type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    deprecated: true
+    deprecationWarning: v1beta2 ImageUpdateAutomation is deprecated, upgrade to v1
     name: v1beta2
     schema:
       openAPIV3Schema:
@@ -430,6 +536,7 @@
                         description: |-
                           MessageTemplate provides a template for the commit message,
                           into which will be interpolated the details of the change made.
+                          Note: The `Updated` template field has been removed. Use `Changed` instead.
                         type: string
                       messageTemplateValues:
                         additionalProperties:
@@ -597,8 +704,6 @@
                     enum:
                     - Setters
                     type: string
-                required:
-                - strategy
                 type: object
             required:
             - interval
@@ -721,7 +826,7 @@
             type: object
         type: object
     served: true
-    storage: true
+    storage: false
     subresources:
       status: {}
 {{- end }}
diff -ru a/bitnami/flux/templates/crds/image-reflector-controller-crds.yaml b/bitnami/flux/templates/crds/image-reflector-controller-crds.yaml
--- a/bitnami/flux/templates/crds/image-reflector-controller-crds.yaml	2026-02-20 01:24:48
+++ b/bitnami/flux/templates/crds/image-reflector-controller-crds.yaml	2026-02-20 01:29:46
@@ -1,13 +1,14 @@
 # Source: https://github.com/fluxcd/image-reflector-controller/releases/download/v{version}/image-reflector-controller.crds.yaml
-# Version: 0.35.2
+# Version: 1.1.0
 # VersionOf: fluxcd-image-reflector-controller
 # Conditional: .Values.imageReflectorController.installCRDs
 {{- if .Values.imageReflectorController.installCRDs }}
+---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.16.1
+    controller-gen.kubebuilder.io/version: v0.19.0
   name: imagepolicies.image.toolkit.fluxcd.io
 spec:
   group: image.toolkit.fluxcd.io
@@ -15,14 +16,29 @@
     kind: ImagePolicy
     listKind: ImagePolicyList
     plural: imagepolicies
+    shortNames:
+    - imgpol
+    - imagepol
     singular: imagepolicy
   scope: Namespaced
   versions:
   - additionalPrinterColumns:
-    - jsonPath: .status.latestImage
-      name: LatestImage
+    - jsonPath: .status.latestRef.name
+      name: Image
       type: string
-    name: v1beta1
+    - jsonPath: .status.latestRef.tag
+      name: Tag
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].message
+      name: Status
+      type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v1
     schema:
       openAPIV3Schema:
         description: ImagePolicy is the Schema for the imagepolicies API
@@ -47,8 +63,27 @@
           spec:
             description: |-
               ImagePolicySpec defines the parameters for calculating the
-              ImagePolicy
+              ImagePolicy.
             properties:
+              digestReflectionPolicy:
+                default: Never
+                description: |-
+                  DigestReflectionPolicy governs the setting of the `.status.latestRef.digest` field.
+
+                  Never: The digest field will always be set to the empty string.
+
+                  IfNotPresent: The digest field will be set to the digest of the elected
+                  latest image if the field is empty and the image did not change.
+
+                  Always: The digest field will always be set to the digest of the elected
+                  latest image.
+
+                  Default: Never.
+                enum:
+                - Always
+                - IfNotPresent
+                - Never
+                type: string
               filterTags:
                 description: |-
                   FilterTags enables filtering for only a subset of tags based on a set of
@@ -81,6 +116,15 @@
                 required:
                 - name
                 type: object
+              interval:
+                description: |-
+                  Interval is the length of time to wait between
+                  refreshing the digest of the latest tag when the
+                  reflection policy is set to "Always".
+
+                  Defaults to 10m.
+                pattern: ^([0-9]+(\.[0-9]+)?(ms|s|m|h))+$
+                type: string
               policy:
                 description: |-
                   Policy gives the particulars of the policy to be followed in
@@ -130,10 +174,24 @@
                     - range
                     type: object
                 type: object
+              suspend:
+                description: |-
+                  This flag tells the controller to suspend subsequent policy reconciliations.
+                  It does not apply to already started reconciliations. Defaults to false.
+                type: boolean
             required:
             - imageRepositoryRef
             - policy
             type: object
+            x-kubernetes-validations:
+            - message: spec.interval is only accepted when spec.digestReflectionPolicy
+                is set to 'Always'
+              rule: '!has(self.interval) || (has(self.digestReflectionPolicy) && self.digestReflectionPolicy
+                == ''Always'')'
+            - message: spec.interval must be set when spec.digestReflectionPolicy
+                is set to 'Always'
+              rule: has(self.interval) || !has(self.digestReflectionPolicy) || self.digestReflectionPolicy
+                != 'Always'
           status:
             default:
               observedGeneration: -1
@@ -195,25 +253,76 @@
                   - type
                   type: object
                 type: array
-              latestImage:
+              lastHandledReconcileAt:
                 description: |-
-                  LatestImage gives the first in the list of images scanned by
-                  the image repository, when filtered and ordered according to
-                  the policy.
+                  LastHandledReconcileAt holds the value of the most recent
+                  reconcile request value, so a change of the annotation value
+                  can be detected.
                 type: string
+              latestRef:
+                description: |-
+                  LatestRef gives the first in the list of images scanned by
+                  the image repository, when filtered and ordered according
+                  to the policy.
+                properties:
+                  digest:
+                    description: Digest is the image's digest.
+                    type: string
+                  name:
+                    description: Name is the bare image's name.
+                    type: string
+                  tag:
+                    description: Tag is the image's tag.
+                    type: string
+                required:
+                - name
+                - tag
+                type: object
               observedGeneration:
                 format: int64
                 type: integer
+              observedPreviousRef:
+                description: |-
+                  ObservedPreviousRef is the observed previous LatestRef. It is used
+                  to keep track of the previous and current images.
+                properties:
+                  digest:
+                    description: Digest is the image's digest.
+                    type: string
+                  name:
+                    description: Name is the bare image's name.
+                    type: string
+                  tag:
+                    description: Tag is the image's tag.
+                    type: string
+                required:
+                - name
+                - tag
+                type: object
             type: object
         type: object
     served: true
-    storage: false
+    storage: true
     subresources:
       status: {}
   - additionalPrinterColumns:
-    - jsonPath: .status.latestImage
-      name: LatestImage
+    - jsonPath: .status.latestRef.name
+      name: Image
       type: string
+    - jsonPath: .status.latestRef.tag
+      name: Tag
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].message
+      name: Status
+      type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    deprecated: true
+    deprecationWarning: v1beta2 ImagePolicy is deprecated, upgrade to v1
     name: v1beta2
     schema:
       openAPIV3Schema:
@@ -350,6 +459,11 @@
                     - range
                     type: object
                 type: object
+              suspend:
+                description: |-
+                  This flag tells the controller to suspend subsequent policy reconciliations.
+                  It does not apply to already started reconciliations. Defaults to false.
+                type: boolean
             required:
             - imageRepositoryRef
             - policy
@@ -424,13 +538,11 @@
                   - type
                   type: object
                 type: array
-              latestImage:
+              lastHandledReconcileAt:
                 description: |-
-                  LatestImage gives the first in the list of images scanned by
-                  the image repository, when filtered and ordered according to
-                  the policy.
-
-                  Deprecated: Replaced by the composite "latestRef" field.
+                  LastHandledReconcileAt holds the value of the most recent
+                  reconcile request value, so a change of the annotation value
+                  can be detected.
                 type: string
               latestRef:
                 description: |-
@@ -454,13 +566,6 @@
               observedGeneration:
                 format: int64
                 type: integer
-              observedPreviousImage:
-                description: |-
-                  ObservedPreviousImage is the observed previous LatestImage. It is used
-                  to keep track of the previous and current images.
-
-                  Deprecated: Replaced by the composite "observedPreviousRef" field.
-                type: string
               observedPreviousRef:
                 description: |-
                   ObservedPreviousRef is the observed previous LatestRef. It is used
@@ -482,15 +587,16 @@
             type: object
         type: object
     served: true
-    storage: true
+    storage: false
     subresources:
       status: {}
 ---
+---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.16.1
+    controller-gen.kubebuilder.io/version: v0.19.0
   name: imagerepositories.image.toolkit.fluxcd.io
 spec:
   group: image.toolkit.fluxcd.io
@@ -498,17 +604,33 @@
     kind: ImageRepository
     listKind: ImageRepositoryList
     plural: imagerepositories
+    shortNames:
+    - imgrepo
+    - imagerepo
     singular: imagerepository
   scope: Namespaced
   versions:
   - additionalPrinterColumns:
-    - jsonPath: .status.lastScanResult.scanTime
-      name: Last scan
+    - jsonPath: .spec.image
+      name: Image
       type: string
     - jsonPath: .status.lastScanResult.tagCount
       name: Tags
       type: string
-    name: v1beta1
+    - jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].message
+      name: Status
+      type: string
+    - jsonPath: .status.lastScanResult.scanTime
+      name: Last scan
+      priority: 1
+      type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v1
     schema:
       openAPIV3Schema:
         description: ImageRepository is the Schema for the imagerepositories API
@@ -564,17 +686,21 @@
                 type: object
               certSecretRef:
                 description: |-
-                  CertSecretRef can be given the name of a secret containing
+                  CertSecretRef can be given the name of a Secret containing
                   either or both of
 
-                   - a PEM-encoded client certificate (`certFile`) and private
-                   key (`keyFile`);
-                   - a PEM-encoded CA certificate (`caFile`)
+                  - a PEM-encoded client certificate (`tls.crt`) and private
+                  key (`tls.key`);
+                  - a PEM-encoded CA certificate (`ca.crt`)
 
-                   and whichever are supplied, will be used for connecting to the
-                   registry. The client cert and key are useful if you are
-                   authenticating with a certificate; the CA cert is useful if
-                   you are using a self-signed server certificate.
+                  and whichever are supplied, will be used for connecting to the
+                  registry. The client cert and key are useful if you are
+                  authenticating with a certificate; the CA cert is useful if
+                  you are using a self-signed server certificate. The Secret must
+                  be of type `Opaque` or `kubernetes.io/tls`.
+
+                  Note: Support for the `caFile`, `certFile` and `keyFile` keys has
+                  been deprecated.
                 properties:
                   name:
                     description: Name of the referent.
@@ -583,21 +709,50 @@
                 - name
                 type: object
               exclusionList:
+                default:
+                - ^.*\.sig$
                 description: |-
                   ExclusionList is a list of regex strings used to exclude certain tags
                   from being stored in the database.
                 items:
                   type: string
+                maxItems: 25
                 type: array
               image:
                 description: Image is the name of the image repository
                 type: string
+              insecure:
+                description: Insecure allows connecting to a non-TLS HTTP container
+                  registry.
+                type: boolean
               interval:
                 description: |-
                   Interval is the length of time to wait between
                   scans of the image repository.
                 pattern: ^([0-9]+(\.[0-9]+)?(ms|s|m|h))+$
                 type: string
+              provider:
+                default: generic
+                description: |-
+                  The provider used for authentication, can be 'aws', 'azure', 'gcp' or 'generic'.
+                  When not specified, defaults to 'generic'.
+                enum:
+                - generic
+                - aws
+                - azure
+                - gcp
+                type: string
+              proxySecretRef:
+                description: |-
+                  ProxySecretRef specifies the Secret containing the proxy configuration
+                  to use while communicating with the container registry.
+                properties:
+                  name:
+                    description: Name of the referent.
+                    type: string
+                required:
+                - name
+                type: object
               secretRef:
                 description: |-
                   SecretRef can be given the name of a secret containing
@@ -708,14 +863,36 @@
               lastScanResult:
                 description: LastScanResult contains the number of fetched tags.
                 properties:
+                  latestTags:
+                    description: |-
+                      LatestTags is a small sample of the tags found in the last scan.
+                      It's the first 10 tags when sorting all the tags in descending
+                      alphabetical order.
+                    items:
+                      type: string
+                    type: array
+                  revision:
+                    description: Revision is a stable hash of the scanned tags.
+                    type: string
                   scanTime:
+                    description: ScanTime is the time when the last scan was performed.
                     format: date-time
                     type: string
                   tagCount:
+                    description: TagCount is the number of tags found in the last
+                      scan.
                     type: integer
                 required:
                 - tagCount
                 type: object
+              observedExclusionList:
+                description: |-
+                  ObservedExclusionList is a list of observed exclusion list. It reflects
+                  the exclusion rules used for the observed scan result in
+                  spec.lastScanResult.
+                items:
+                  type: string
+                type: array
               observedGeneration:
                 description: ObservedGeneration is the last reconciled generation.
                 format: int64
@@ -723,16 +900,31 @@
             type: object
         type: object
     served: true
-    storage: false
+    storage: true
     subresources:
       status: {}
   - additionalPrinterColumns:
-    - jsonPath: .status.lastScanResult.scanTime
-      name: Last scan
+    - jsonPath: .spec.image
+      name: Image
       type: string
     - jsonPath: .status.lastScanResult.tagCount
       name: Tags
       type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].status
+      name: Ready
+      type: string
+    - jsonPath: .status.conditions[?(@.type=="Ready")].message
+      name: Status
+      type: string
+    - jsonPath: .status.lastScanResult.scanTime
+      name: Last scan
+      priority: 1
+      type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    deprecated: true
+    deprecationWarning: v1beta2 ImageRepository is deprecated, upgrade to v1
     name: v1beta2
     schema:
       openAPIV3Schema:
@@ -967,13 +1159,23 @@
                 description: LastScanResult contains the number of fetched tags.
                 properties:
                   latestTags:
+                    description: |-
+                      LatestTags is a small sample of the tags found in the last scan.
+                      It's the first 10 tags when sorting all the tags in descending
+                      alphabetical order.
                     items:
                       type: string
                     type: array
+                  revision:
+                    description: Revision is a stable hash of the scanned tags.
+                    type: string
                   scanTime:
+                    description: ScanTime is the time when the last scan was performed.
                     format: date-time
                     type: string
                   tagCount:
+                    description: TagCount is the number of tags found in the last
+                      scan.
                     type: integer
                 required:
                 - tagCount
@@ -993,7 +1195,7 @@
             type: object
         type: object
     served: true
-    storage: true
+    storage: false
     subresources:
       status: {}
 {{- end }}
